src/Domain/SizeOfElement.cpp

```diff
@@ -3,37 +3,35 @@
 
 #include "Domain/SizeOfElement.hpp"
 
-#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
-#include "DataStructures/Tensor/TypeAliases.hpp"
-#include "Domain/Mesh.hpp"  // IWYU pragma: keep
+#include "Domain/ElementMap.hpp"  // IWYU pragma: keep
 #include "Domain/Side.hpp"
-#include "ErrorHandling/Assert.hpp"
-#include "NumericalAlgorithms/LinearOperators/MeanValue.hpp"
-#include "NumericalAlgorithms/Spectral/Spectral.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/MakeArray.hpp"
 #include "Utilities/StdArrayHelpers.hpp"  // IWYU pragma: keep
 
 template <size_t VolumeDim>
 std::array<double, VolumeDim> size_of_element(
-    const Mesh<VolumeDim>& mesh,
-    const tnsr::I<DataVector, VolumeDim>& inertial_coords) noexcept {
-  ASSERT(mesh.quadrature() ==
-             make_array<VolumeDim>(Spectral::Quadrature::GaussLobatto),
-         "Implementation assumes that grid points extend to the faces of the\n"
-         "element, but not sure if this is true for quadrature: "
-             << mesh.quadrature());
+    const ElementMap<VolumeDim, Frame::Inertial>& element_map) noexcept {
   auto result = make_array<VolumeDim>(0.0);
-  // inertial-coord vector between lower face center and upper face center
-  auto center_to_center = make_array<VolumeDim>(0.0);
   for (size_t logical_dim = 0; logical_dim < VolumeDim; ++logical_dim) {
+    const auto face_center =
+        [&logical_dim, &element_map ](const Side& side) noexcept {
+      tnsr::I<double, VolumeDim, Frame::Logical> logical_center{{{0.0}}};
+      logical_center.get(logical_dim) = (side == Side::Lower ? -1.0 : 1.0);
+      const tnsr::I<double, VolumeDim, Frame::Inertial> inertial_center =
+          element_map(logical_center);
+      return inertial_center;
+    };
+    const auto lower_center = face_center(Side::Lower);
+    const auto upper_center = face_center(Side::Upper);
+
+    // inertial-coord distance from lower face center to upper face center
+    auto center_to_center = make_array<VolumeDim>(0.0);
```

Comment:
I think you can just subtract the arrays if you include `"Utilities/StdArrayHelpers.hpp"`

Reply:
`upper_center` and `lower_center` are Tensor objects, and afaik I can't subtract them directly

Reply:
so they are, never mind...